### DISCUSSION TITLE
LibraryState and its state_copy should have their own port elements

### DISCUSSION THREAD

**Discussion by franzlst** [DLR-RM/RAFCON#962] [General]
A LibraryState and its `state_copy` share the same port core elements. This should be avoided, as with this, the ports, which are leave elements, have two different parents. This causes problems with the GraphicalEditor.

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2018-02-20 08:32:55+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I think similar  as @sebastian-brunner already named it once, duplicating something which should be identically should not be the solution for GUI problems.

In fact the state copy is the only parent of the ports and the library state is overlaying it. What I would prefer to simplify it and maybe get both more like on object and the library state class more like inherit from the state copy class, to have more clear and direct dependencies to avoid understanding problems in the future.

Basically there are two options one is to make this both elements closer to each other or to separate both very strongly. 

We can bring them e.g. closer if we make thinks which are implicit dependent more explicit what will make the dependencies transparent.

If we separate both we will run into the problem that we create at least another minor or?

Originally created by @Rbelder  at 2018-02-20 17:57:25+00:00 (moved from RMC internal repository)
